### ISSUE TITLE
Nested to_anf Calls Produce Incorrect Results

### ISSUE THREAD

**Issue by PhysicsQoo** [sympy/sympy#26985]
I encountered an issue when using nested to_anf calls in SymPy, which leads to incorrect and unexpected results. This issue is particularly concerning in scenarios where precise Boolean algebra manipulations are critical, and nested operations might be used to incrementally simplify or modify expressions.

# Steps to Reproduce:
Here’s a minimal example to illustrate the issue:

```
from sympy import symbols, And, Xor
from sympy.logic.boolalg import to_anf

a, b, c, d= symbols('a b c d')

# Expression before applying to_anf
x = Xor(c, And(a, b), And(a, c))
y = Xor(a, b, And(a, c))

# Applying to_anf
result =Xor(Xor(d, And(x, y)), And(x, y))
result_anf = to_anf(Xor(to_anf(Xor(d, And(x, y))), And(x, y)))

print(f'result: {result}')
print(f'result_anf: {result_anf}')
```
output:
```
result: d
result_anf: d ^ (b & c)
```
# Expected Behavior:
Given the Boolean logic, the original expression simplifies to 'd'. 
The expected output after calling to_anf should also be 'd'

Instead, the nested to_anf calls produce the following incorrect result: d ^ (b & c)

# Analysis:

- When nesting to_anf calls, it seems that certain terms are reintroduced or incorrectly simplified, which leads to the generation of extra terms that should not be present in the output.

- This behavior suggests that the conversion process within to_anf does not fully account for nested or partially simplified expressions, leading to incorrect results.

# Environment:
- SymPy version: 1.12
- Python version: 3.12.1

**Comment by oscarbenjamin** [comment]
This seems to be already fixed on master:
```python
In [1]: from sympy import symbols, And, Xor
   ...: from sympy.logic.boolalg import to_anf
   ...: 
   ...: a, b, c, d= symbols('a b c d')
   ...: 
   ...: # Expression before applying to_anf
   ...: x = Xor(c, And(a, b), And(a, c))
   ...: y = Xor(a, b, And(a, c))
   ...: 
   ...: # Applying to_anf
   ...: result =Xor(Xor(d, And(x, y)), And(x, y))
   ...: result_anf = to_anf(Xor(to_anf(Xor(d, And(x, y))), And(x, y)))
   ...: 
   ...: print(f'result: {result}')
   ...: print(f'result_anf: {result_anf}')
result: d
result_anf: d
```

I've bisected the fix to da5025954d1880652775eed384b3744c258755b9 from gh-26183.

If you update to SymPy 1.13 then you should see that it is fixed.

It would be good to add this example to the tests so this issue can be closed if a test case is added.

**Comment by oscarbenjamin** [comment]
Fixed by gh-27017
